### PR TITLE
new package: ko-fips

### PR DESCRIPTION
**Pull request by imjasonh**
This is mainly a trial run to get an idea for how we can build FIPS-compliant Go packages and images, starting with a Go application we know and understand well.

Once we've proven this out we can move on to more interesting FIPSified Go applications.

### CODE REVIEW

**Comment by imjasonh** [comment]
`go-fips` conflicts with the builtin `go/build` pipeline, good to know. ⭐

**Comment by mattmoor** [comment]
What's the best option?  `go-fips/build`? :)

**Comment by imjasonh** [comment]
> What's the best option? `go-fips/build`? :)

For now I'm just going to do it with `runs:`, but yeah, we might consider having a go-fips variant, probably only in the wolfi-os repo, rather than built into melange like `go/build` is.

**Review by kaniini — DISMISSED** [review]

File: ko-fips.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      repository: https://github.com/ko-build/ko
+      tag: v${{package.version}}
+      expected-commit: 200db7243f02b5c0303e21d8ab8e3b4ad3a229d0
+      destination: ko
+
+  - runs: |
+      cd ko
+      go build -o "${{targets.destdir}}"/usr/bin/ko \
+          -ldflags "-s -w -X github.com/google/ko/pkg/commands.Version=${{package.version}}" \
+          -trimpath \
```
Comment (mattmoor) [inline ko-fips.yaml:30]: extra trailing slash

File: ko-fips.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package:
+  name: ko-fips
+  version: 0.14.1
```
Comment (mattmoor) [inline ko-fips.yaml:3]:
```suggestion
  version: 0.14.1
```

**Review by luhring — APPROVED** [review]

**Pull request merged**